packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol

```diff
@@ -474,15 +475,39 @@ contract PerpetualPositionManager is FeePayer, AdministrateeInterface {
         tokenCurrency.burn(numTokens.rawValue);
     }
 
+    /**
+     * @notice Burns `numTokens` of `tokenCurrency` to decrease sponsors position size, without sending back any
+     * `collateralCurrency`. This is done by a sponsor to increase position CR.
+     * @dev Can only be called by token sponsor. This contract must be approved to spend `numTokens` of `tokenCurrency`.
+     * @param numTokens is the number of tokens to be burnt for a commensurate amount of collateral.
+     */
     function repay(FixedPoint.Unsigned memory numTokens)
         public
         notEmergencyShutdown()
         noPendingWithdrawal(msg.sender)
         fees()
         nonReentrant()
-        returns (FixedPoint.Unsigned memory amountRepaired)
     {
-        // TODO: implementation
+        PositionData storage positionData = _getPositionData(msg.sender);
+        require(!numTokens.isGreaterThan(positionData.tokensOutstanding), "Invalid token amount");
```

Comment:
I feel like this is slightly more readable as numTokens.isLessThanOrEqualTo

Reply:
I would agree but I did this to be consistent with the implementation of similar checks in `redeem` which can be seen here: https://github.com/UMAprotocol/protocol/blob/3129fc83480531d31ba64af5bb10ac061f93e6bf/packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol#L446 this logic was present in the previous EMP design.

If we were to change this then we should change it throught. thoughts?

Reply:
Hm, it's not a huge deal but I think using positive booleans rather than negating booleans is more readable. As long as it doesn't create any additional churn I'd recommend changing both to the positive `isLessThanOrEuqlaTo`

Reply:
Yeah, the only reason some of them use the `!` syntax is because we used to not have `OrEqualTo` functions I think.

Reply:
Cool & done. There were two other places in this contract that also required this change.